src/zenml/integrations/skypilot/orchestrators/skypilot_base_vm_orchestrator.py

```diff
@@ -0,0 +1,252 @@
+
+if TYPE_CHECKING:
+    from zenml.models.pipeline_deployment_models import (
+        PipelineDeploymentResponseModel,
+    )
+    from zenml.stack import Stack
+
+
+logger = get_logger(__name__)
+
+ENV_ZENML_SKYPILOT_ORCHESTRATOR_RUN_ID = "ZENML_SKYPILOT_ORCHESTRATOR_RUN_ID"
+
+
+class SkypilotBaseOrchestrator(ContainerizedOrchestrator):
+    """Base class for Orchestrator responsible for running pipelines remotely in a VM.
+
+    This orchestrator does not support running on a schedule.
+    """
+
+    # The default instance type to use if none is specified in settings
+    DEFAULT_INSTANCE_TYPE: Optional[str] = None
+
+    @property
+    def validator(self) -> Optional[StackValidator]:
+        """Validates the stack.
+
+        In the remote case, checks that the stack contains a container registry,
+        image builder and only remote components.
+
+        Returns:
+            A `StackValidator` instance.
+        """
+
+        def _validate_remote_components(
+            stack: "Stack",
+        ) -> Tuple[bool, str]:
+            for component in stack.components.values():
+                if not component.config.is_local:
+                    continue
+
+                return False, (
+                    f"The Skypilot orchestrator runs pipelines remotely, "
+                    f"but the '{component.name}' {component.type.value} is "
+                    "a local stack component and will not be available in "
+                    "the Skypilot step.\nPlease ensure that you always "
+                    "use non-local stack components with the Skypilot "
+                    "orchestrator."
+                )
+
+            return True, ""
+
+        return StackValidator(
+            required_components={
+                StackComponentType.CONTAINER_REGISTRY,
+                StackComponentType.IMAGE_BUILDER,
+            },
+            custom_validation_function=_validate_remote_components,
+        )
+
+    def get_orchestrator_run_id(self) -> str:
+        """Returns the active orchestrator run id.
+
+        Raises:
+            RuntimeError: If the environment variable specifying the run id
+                is not set.
+
+        Returns:
+            The orchestrator run id.
+        """
+        try:
+            return os.environ[ENV_ZENML_SKYPILOT_ORCHESTRATOR_RUN_ID]
+        except KeyError:
+            raise RuntimeError(
+                "Unable to read run id from environment variable "
+                f"{ENV_ZENML_SKYPILOT_ORCHESTRATOR_RUN_ID}."
+            )
+
+    @property
+    @abstractmethod
+    def cloud(self) -> sky.clouds.Cloud:
+        """The type of sky cloud to use.
+
+        Returns:
+            A `sky.clouds.Cloud` instance.
+        """
+
+    def setup_credentials(self) -> None:
+        """Set up credentials for the orchestrator."""
+        connector = self.get_connector()
+        assert connector is not None
+        connector._configure_local_client()
```

Comment:
```suggestion
        connector.configure_local_client()
```

The API method has some extra steps, like validating the configuration and credentials.